src/robotlibcore.py

```diff
@@ -142,6 +135,38 @@ def get_keyword_documentation(self, name):
             doc = '{}\n\n{}'.format(doc, tags) if doc else tags
         return doc
 
+    def get_keyword_types(self, keyword_name):
+        method = self.keywords.get(keyword_name)
+        if keyword_name == '__init__':
+            method = self.__init__
+        elif keyword_name.startswith('__') and keyword_name.endswith('__'):
+            return {}
+        if not method:
+            raise ValueError('%s is not keyword.' % keyword_name)
+        robot_types = getattr(method, 'robot_types', ())
+        if robot_types is None:
+            return robot_types
+        annotations = self._get_annotations(method)
+        types = self._join_defaults_with_annotations(method, annotations)
+        if robot_types:
+            types.update(robot_types)
+        return types
+
+    def _get_annotations(self, method):
+        if PY2:
+            return {}
+        annotations = typing.get_type_hints(method)
+        annotations.pop('return', None)
+        return annotations
+
+    def _join_defaults_with_annotations(self, method, annotations):
+        _, defaults, _, _ = self._get_arg_spec(method)
+        for default in defaults:
+            if default[1] is False or default[1] is True or default[1] is None:
+                if default[0] not in annotations:
+                    annotations[default[0]] = type(default[1])
```

Comment:
1. This should be used with types got from `@keyword` if they are defined and with type hints otherwise. Thus `annotations` should be named `types` in method and variable names.

2. The loop can be still simplified

    ```python
    for name, value in defaults:
        if name not in types and isinstance(value, (bool, type(None)):
            types[name] = type(value)
    ```